File: iScoreImport/iScoreImport/Models/Pitcher.swift
//
//  Pitcher.swift
//  iScoreImport
//
//  Created by Patrick Eschenfeldt on 7/4/24.
//

import Foundation

struct Pitcher: Codable {
    let PlayerExternalId: UUID
    let PlayerName: String
    let TeamExternalId: UUID
    
    let Number: Int
    let Games: Int
    let Wins: Int
    let Losses: Int
    let Saves: Int
    let ThirdInningsPitched: Int
    let BattersFaced: Int
    let Balls: Int
    let Strikes: Int
    let Runs: Int
    let EarnedRuns: Int
    let Hits: Int
    let Walks: Int
    let IntentionalWalks: Int
    let Strikeouts: Int
    let StrikeoutsCalled: Int
    let StrikeoutsSwinging: Int
    let HitByPitch: Int
    let Balks: Int
    let WildPitches: Int
    let Homeruns: Int
    let GroundOuts: Int
    let AirOuts: Int
    let FirstPitchStrikes: Int
    let FirstPitchBalls: Int
}

File: BaseballApi/Models/Pitcher.cs
using System.ComponentModel;
using Microsoft.EntityFrameworkCore;

namespace BaseballApi.Models;

[Index(nameof(BoxScoreId))]
[Index(nameof(BoxScoreId), nameof(PlayerId), IsUnique = true)]
public class Pitcher
{
    public long Id { get; set; }

    public long BoxScoreId { get; set; }
    public required BoxScore BoxScore { get; set; }

    public long PlayerId { get; set; }
    public required Player Player { get; set; }

    public int Number { get; set; }
    public int Games { get; set; }
    public int Wins { get; set; }
    public int Losses { get; set; }
    public int Saves { get; set; }
    [Description("Innings Pitched times 3; i.e. outs recorded")]
    public int ThirdInningsPitched { get; set; }
    public int BattersFaced { get; set; }
    public int Balls { get; set; }
    public int Strikes { get; set; }
    public int Pitches { get; set; }
    public int Runs { get; set; }
    public int EarnedRuns { get; set; }
    public int Hits { get; set; }
    public int Walks { get; set; }
    public int IntentionalWalks { get; set; }
    public int Strikeouts { get; set; }
    public int StrikeoutsCalled { get; set; }
    public int StrikeoutsSwinging { get; set; }
    public int HitByPitch { get; set; }
    public int Balks { get; set; }
    public int WildPitches { get; set; }
    public int Homeruns { get; set; }
    public int GroundOuts { get; set; }
    public int AirOuts { get; set; }
    public int FirstPitchStrikes { get; set; }
    public int FirstPitchBalls { get; set; }
}
